packages/starlight/user-components/LinkButton.astro

```diff
@@ -0,0 +1,72 @@
+---
+import type { HTMLAttributes } from 'astro/types';
+import { Icons } from '../components/Icons';
+import Icon from './Icon.astro';
+
+interface Props extends Omit<HTMLAttributes<'a'>, 'href'> {
+	href: string | URL;
+	icon?: keyof typeof Icons | undefined;
+	iconPlacement?: 'start' | 'end' | undefined;
+	variant?: 'primary' | 'secondary' | 'minimal';
+}
+
+const {
+	class: className,
+	icon,
+	iconPlacement = 'end',
+	variant = 'primary',
+	...attrs
+} = Astro.props;
+---
+
+<a class:list={['sl-link-button not-content', variant, className]} {...attrs}>
+	{icon && iconPlacement === 'start' && <Icon name={icon} size="1.5rem" />}
+	<slot />
+	{icon && iconPlacement === 'end' && <Icon name={icon} size="1.5rem" />}
+</a>
+
+<style>
+	.sl-link-button {
```

Comment:
Nit: the default new component is 2px taller than the current one (58px vs 56px on the home page in desktop view, 42px vs 40px in mobile view). That sounds like not much, but I definitely feel it as a little bit chunkier, so would be great to keep the height consistent with the current appearance.

Reply:
Well, I don't think this is technically true. Current ones are `56px` (`40px` on mobile) if you only use primary and minimal variants. As soon as you introduce the secondary variant, the height becomes `58px` (`42px` on mobile). This PR makes the heights consistent across all variants to avoid the sizing inconsistencies you described in the previous comment. Altho, I picked `58px` as the height for all variants, but we can definitely change this to `56px` if you think that would be better but wanted to make sure you are aware of this before I make the change.

Reply:
Ah yeah, good point. Yes, let’s standardise to 56px — the [design reference in Figma](https://www.figma.com/design/GrDkcguAR7tSWPFuLVDixq/Theme---Starlight-Docs?node-id=171-506&t=Mo8nQ4EThzphRaE4-4) is also 56px for all variants (thanks to Figma’s utopian “inside” stroke 😁).